persist/fs/files.go

```diff
@@ -42,11 +42,117 @@ var timeZero time.Time
 
 const (
 	dataDirName       = "data"
+	snapshotDirName   = "snapshots"
 	commitLogsDirName = "commitlogs"
 )
 
 type fileOpener func(filePath string) (*os.File, error)
 
+// FilesetFile represents a set of Fileset files for a given block start
+type FilesetFile struct {
+	BlockStart        time.Time
+	AbsoluteFilepaths []string
+}
+
+// FilesetFilesSlice is a slice of FilesetFile
+type FilesetFilesSlice []FilesetFile
+
+// Filepaths flattens a slice of FilesetFiles to a single slice of filepaths.
+// All paths returned are absolute.
+func (f FilesetFilesSlice) Filepaths() []string {
+	flattened := []string{}
+	for _, fileset := range f {
+		flattened = append(flattened, fileset.AbsoluteFilepaths...)
+	}
+
+	return flattened
+}
+
+// NewFilesetFile creates a new Fileset file
+func NewFilesetFile(blockStart time.Time) FilesetFile {
+	return FilesetFile{
+		BlockStart:        blockStart,
+		AbsoluteFilepaths: []string{},
+	}
+}
+
+// SnapshotFile represents a set of Snapshot files for a given block start
+type SnapshotFile struct {
+	// There can be multiple snapshot files with the same blockstart, so they
```

Comment:
ordering nit: put the embedded structs first, then the fields you're adding

Reply:
Agreed

Reply:
resolved cause I'm not adding any new fields anymore